### ISSUE TITLE
Answer format odd for Incr/Decr questions (Section 3.3)

### ISSUE THREAD

**Issue by APEXCalculus** [APEXCalculus/APEXCalculusPTX#245]
The answers to #15 - 21 have odd formatting. The question has 4 parts, a - d, but the answers are given as Answer 1 ... Answer 6. The intervals for incr/decr are being split into separate answers, I think, increasing the number of answers.

**Comment by Alex-Jordan** [comment]
Formerly, an exercise could only artificially be broken into parts using a list. This cannot be recognized as actually breaking the exercise into parts, since it's reasonable to simply be printing a list in an exercise.

In modern times, such an exercise should use "task" elements for the "a", "b", "c", "d". Each task will have its own areas for hint, answer, solution. Some tasks in these exercises apparently have more than one answer blank, so you'll get something like an "Answer 1" and and "Answer 2" within such a task, with the default labeling.

So all these multipart questions throughout the book should be converted to use "task". What fun!

**Comment by sean-fitzpatrick** [comment]
Converting examples and exercises to use tasks was on my to-do list last summer, but then I realized the job was too big to complete before I ran out of summer.

The same may be true this summer, I'm afraid...

That said: this is higher priority for exercises than it is for examples. @APEXCalculus if you have time to compile a list of exercises that have parts, I will go through those and update the code to use tasks.

**Comment by sean-fitzpatrick** [comment]
@Alex-Jordan there are at least a few places where the `<ol>` is in the `exercisegroup` introduction, rather than in the problem itself. What is the best way to proceed with these?

See, for example: https://github.com/APEXCalculus/APEXCalculusPTX/blob/9dc5ee9695094fe341d74b8d02762df41da4fde0/ptx/sec_ABC.ptx#L1482

**Comment by Alex-Jordan** [comment]
I don't know about other examples, but here it's asking for one answer, but with evidence that you find it in two different ways. So I would leave the `ol` in the introduction. Then have one answer blank for the area. And then use an "explanation essay" answer field for showing the work for doing it thee two ways. Search the repo for `showwork` to find examples of how to put in an explanation essay field.

**Comment by sean-fitzpatrick** [comment]
OK. In cases where two answers are expected, I guess I can structure those with tasks, with minimal instructions for each part.

That particular problem will have to wait for a different pull request, since it's one of many where the WeBWorK code was commented out while waiting for TikZ infrastructure, which we now have.

I'm not sure if you have time for that this summer. I can try to get to it, but my to-do list seems to be gaining items faster than I can cross them off!

**Comment by sean-fitzpatrick** [comment]
@APEXCalculus I would like your opinion on how best to proceed.

Currently, these questions (and a number of others) are structured something like the following:

**Exercise group**. For the functions below,
a. Do the first thing
b. Do the second thing
c. Do the third thing

11. Some function
12. Another function
13. Yet another function
14. A new function


To make these work as interactive exercises, @Alex-Jordan structured the code for each problem as follows:

f(x) = some function
<instruction>
Do the first thing
</instruction>
____Answer blank for the first thing
<instruction>
Do the second thing
</instruction>
____Answer blank for the second thing
<instruction>
Do the third thing
</instruction>
____Answer blank for the third thing

The reason for doing it this way is that all of the instructions and answer blanks are invisible in "static" versions, including PDF, and HTML prior to activation. But once a student activates a question (or if they are doing this as part of separate online homework), the instructions show up so that they know what to put in each answer blank.

The downside, as you have noticed, is that when you look at the answers in the back, instead of answers for a, b, c you see Answer1 Answer2 Answer3.

In some cases there are more answers than parts. This is because an instruction like "construct the number line to show the intervals of increase and decrease" is difficult (if not impossible) to implement on WeBWorK.

But it is relatively easy to split this into two parts: first, "list the intervals where f is increasing", and second, "list the intervals where f is decreasing".


I can restructure these exercises to avoid this issue. But the cost of fixing the answers in the back will be adding clutter to the exercises themselves. If I restructure using `<task>`, things would look more like this:

**Exercise group**. For each function below, find the critical points and intervals of increase and decrease.

11. f(x) = some function
     a. Find the critical points of f
     b. Find the intervals on which f is increasing
     c. Find the intervals on which f is decreasing
12. f(x) = some other function
     a. Find the critical points of f
     b. Find the intervals on which f is increasing
     c. Find the intervals on which f is decreasing

In other words, those instructions would be repeated for every exercise, instead of just being given once at the beginning of the exercise group.

I don't think there's a way to get your answers in the back how you want without adding the extra clutter to the exercises.

**Comment by sean-fitzpatrick** [comment]
The pretext `task` provides a way to break an exercise into parts.
The reason to use it instead of `ol` is that each `task` gets its own statement, answer, hint, and solution.

But this isn't as good when you've got a situation like this section (and the next one) where you have many problems of the same type, and you only want to list the instructions once.

We can already repeat the instructions once the question is made interactive without affecting how things appear in static formats. The main problem is that we need to provide an answer blank for each part, and WeBWorK reports each answer back.

As a first step, I can rewrite the instructions in the exercisegroup introduction to match how things are organized for WeBWorK (so there are as many parts as there are answer blanks).

I am not sure if we can change the numering from 1, 2, 3,... to a, b, c,...
That is a question for @Alex-Jordan when he is back from holidays.

**Comment by APEXCalculus** [comment]
That seems like the appropriate first step - at least a student's fourth answer will be aligned with the book's fourth answer.

This could be another case of post-processing the latex. I know that isn't what you'd like to hear, but it is an option.

Oh! In my previous comment, I said to see my note below. Technically, this note is below ...
*Note:
Some of the interval answers use union notation. At some point we discussed taking that out, and just giving a list of intervals on which a function is increasing/decr. I see that some of that union notation has survived; see lines 1667, 1755, and 1852 of `sec_graph_incr_decr.ptx`. If I attempt to fix that, I'll break many things, I'm sure.

**Comment by sean-fitzpatrick** [comment]
First, a progress update: the only remaining issue on the answer formatting is that in the exercise group introduction we have a, b, c numbering, and in the back, we have 1, 2, 3. I've started a thread on the pretext development group to see if there's anything I can do about the 1, 2, 3. If not, we could ensure alignment by changing the a, b, c, numbering to 1, 2, 3.


About union notation. We could get rid of it. But we may also have to change wording. The domain, as a set, necessarily involves union. We could say, "list the intervals on which f is defined" instead of "give the domain of f".

Where I think we were in agreement: union should be avoided when answering questions about continuity.
E.g. if f is continuous on [1,2) and on [2,3), that does not mean f is continuous on [1,2)U[2,3)=[1,3).

In that case we are avoiding union for mathematical reasons.

If your preference is to avoid it entirely for pedagogical reasons, that could probably be done.

**Comment by sean-fitzpatrick** [comment]
There are only 13 webwork exercises in 3 files where I find the string `)U(` so it won't be much work to rewrite those to use lists.
(Famous last words...)

Searching for the string `\cup` turns up 26 results in 9 files. We use union notation, for example, when stating domains of inverse trig and inverse hyperbolic functions. That might be harder to get rid of.

**Comment by APEXCalculus** [comment]
Uh oh - I think I messed up here.
My concern was for incr/decr, where in the past I had used union notation
for *sets* on which a function was incr/decr. We had changed the wording of
those problems, and I thought that all was fixed until I looked at some
answers the other day and saw the union symbol. *BUT* I was looking at
domain, not incr/decr. I didn't realize what part of the question I was
looking at. My apologies.

I'm fine leaving things be, and I'm sorry for starting this up!

(In a thread long ago, we did mention that students will not necessarily
have experience with union notation. So instead of writing
(-\infty,1) \cup (1,3) \cup (3,\infty)
we could write something like
all x \in R, except x \neq 1, 3
We could use better set notation, but that also brings the issue that
students won't necessarily know set notation at this point to write
{x \in R | x \neq 1, 3 }.
All that being said, I'm fine leaving things be.)

**Comment by sean-fitzpatrick** [comment]
I think we can close this, unless @Alex-Jordan has thoughts on how to change the numbering format when there are multiple answers from 1, 2, 3 to a, b, c. If not, I will change the numbering in the exercise group introduction to match the 1, 2, 3 in the answers.

**Comment by Alex-Jordan** [comment]
I think it is reasonable for PTX to optionally use a,b,c to enumerate answers. I can't make that feature happen at this time, but Rob might respond to Sean's post about it.

FWIW, it is possible to make a WW answer blank where the following are understood:
- `R - {1,3}`
- `(-infinity, 1) U (1, 3) U (3, infinity)`
- `{x : x != 1 and x!= 3}`  or  `{x | x != 3 and x!= 3}` (but you can't make it understand both at the same time)
- more things, I can't make an exhaustive list

But there is nothing right now that would literally understand `{x in R | x \neq 1, 3 }`. It would need to be entered like one of the above.


PS. You can use `∞` and `≠` directly now, if a student knows how to get those characters.

**Comment by APEXCalculus** [comment]
I'd rather live with the mismatch; I can fix that in the LaTeX manually (or
with the proper regex) later, too.

**Comment by sean-fitzpatrick** [comment]
Thanks. I'll at least try to make sure that the number of steps in the instructions matches the number of answers to be filled in.

**Comment by sean-fitzpatrick** [comment]
OK. With some help from Rob, I can change the numbering of answers to follow a, b, c instead of 1, 2, 3.

I am going to apply this to PDF only -- I don't think too many people bother to visit the answers section in HTML.

**Comment by sean-fitzpatrick** [comment]
This is done. PR coming soon.
